seqr/views/apis/family_api.py

```diff
@@ -133,6 +135,35 @@ def update_family_fields_handler(request, family_guid):
     })
 
 
+@login_required(login_url=API_LOGIN_REQUIRED_URL)
+@csrf_exempt
+def update_family_assigned_analyst(request, family_guid):
+    """Updates the specified field in the Family model.
+
+    Args:
+        family_guid (string): GUID of the family.
+    """
+    family = Family.objects.get(guid=family_guid)
+    check_permissions(family.project, request.user, CAN_EDIT)
+
+    request_json = json.loads(request.body)
+    assigned_analyst_username = request_json.get('assigned_analyst_username')
+
+    if not assigned_analyst_username:
+        return create_json_response(
+            {}, status=400, reason="'assigned analyst' is not specified")
+    try:
+        assigned_analyst = User.objects.get(username=assigned_analyst_username)
+    except Exception:
+        return create_json_response(
+            {}, status=400, reason="user does not have access")
```

Comment:
this isn't really an accurate message, the actual error is "specified user des not exist"